bwapi/bwdata.py

```diff
@@ -937,18 +937,16 @@ def _fill_params(self, name, startDate, data):
 
         for n in name_list:
             if isinstance(n, str):
-                if n not in self.ids:
+                if self.get_resource_id(n, check=True) == False:
```

Comment:
I don't like the semantics of `check` because it changes the behaviour of the method in way that's not obvious.  Instead I'd suggest writing two methods: `get_resource_id` that  does exact what you'd expected, and `resource_name_exists()` which implements the functionality of "check". That would make the above line read:
```python
 if self.resource_exists(name): 
```
which almost reads like English.

Also note I'd suggest rename the `n` to `name` because again, it's easier to read

Reply:
fixed